lib/shopify-cli/core/monorail.rb

```diff
@@ -13,19 +13,23 @@ module Monorail
       @metadata = {}
 
       class << self
-        attr_reader :metadata
+        attr_accessor :metadata
 
         def log(name, args, &block) # rubocop:disable Lint/UnusedMethodArgument
           prompt_for_consent
+          return yield unless enabled? && consented?
+
           args = args.dup.unshift(name)
           start_time = now_in_milliseconds
+          err = nil
           begin
             result = yield
-            send_event(start_time, args)
             return result
           rescue Exception => e # rubocop:disable Lint/RescueException
-            send_event(start_time, args, e.message)
+            err = e
```

Comment:
Yes, I am basically just hoisting the error value. This was honestly just to make the tests better because webmock would raise an error for both the first send_event and then the exception send event (because webmock raised)